src/rules/deprecationRule.ts

```diff
@@ -37,60 +37,79 @@ export class Rule extends Lint.Rules.TypedRule {
         requiresTypeInfo: true,
     };
     /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING(name: string, message: string) {
+        return `${name} is deprecated${message === "" ? "." : `: ${message.trim()}`}`;
+    }
+
     public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
-        return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program));
+        return this.applyWithFunction(sourceFile, (ctx: Lint.WalkContext<void>) => walk(ctx, program.getTypeChecker()));
     }
 }
 
-class Walker extends Lint.ProgramAwareRuleWalker {
-  // Implementation inspired by angular/tsickle:
-  // https://github.com/angular/tsickle/blob/cad7c180a2155db6f6fb8d22c44151d7e8a9149f/src/decorator-annotator.ts#L42
-  protected visitIdentifier(node: ts.Identifier) {
-    let decSym = this.getTypeChecker().getSymbolAtLocation(node);
+function walk(ctx: Lint.WalkContext<void>, tc: ts.TypeChecker) {
+    return ts.forEachChild(ctx.sourceFile, function cb(node): void {
+        if (isIdentifier(node)) {
+            if (!isDeclaration(node)) {
+                const deprecation = getDeprecation(node, tc);
+                if (deprecation !== undefined) {
+                    ctx.addFailureAtNode(node, Rule.FAILURE_STRING(node.text, deprecation));
+                }
+            }
+        } else {
+            return ts.forEachChild(node, cb);
+        }
+    });
+}
 
-    if (decSym !== undefined && Lint.isSymbolFlagSet(decSym, ts.SymbolFlags.Alias)) {
-        decSym = this.getTypeChecker().getAliasedSymbol(decSym);
-    }
-    const declarations = decSym === undefined ? undefined : decSym.getDeclarations() as ts.Node[] | undefined;
-    if (declarations === undefined) {
-        super.visitIdentifier(node);
-        return;
+function isDeclaration(identifier: ts.Identifier): boolean {
```

Comment:
TypeScript has `ts.isDeclarationName`, but it's internal. You could make an issue on the TypeScript issue asking it to be made non-internal since it seems pretty useful.